docs/user/azure/install_upi.md

```diff
@@ -0,0 +1,559 @@
+export BASE_DOMAIN=`yq -r .baseDomain install-config.yaml`
+export BASE_DOMAIN_RESOURCE_GROUP=`yq -r .platform.azure.baseDomainResourceGroupName install-config.yaml`
+export RESOURCE_GROUP=$CLUSTER_NAME
+```
+
+### Empty the compute pool
+
+We'll be providing the compute machines ourselves, so edit the resulting `install-config.yaml` to set `replicas` to 0 for the `compute` pool:
+
+```sh
+python -c '
+import yaml;
+path = "install-config.yaml";
+data = yaml.full_load(open(path));
+data["compute"][0]["replicas"] = 0;
+open(path, "w").write(yaml.dump(data, default_flow_style=False))'
+```
+
+## Create manifests
+
+Create manifests to enable customizations that are not exposed via the install configuration.
+
+```console
+$ openshift-install create manifests
+INFO Credentials loaded from file "/home/user_id/.azure/osServicePrincipal.json"
+INFO Consuming "Install Config" from target directory
+WARNING Making control-plane schedulable by setting MastersSchedulable to true for Scheduler cluster settings
+```
+
+### Customize manifests
+
+The manifests need to reflect the resources to be created by the [Azure Resource Manager][azuretemplates] template, e.g. the
+resource group name. Also, we don't want [the ingress operator][ingress-operator] to create DNS records (we're going to
+do it manually) so we need to remove the `privateZone` and `publicZone` sections from the DNS configuration in manifests.
+
+A Python script named `setup-manifests.py` is provided to help with these and a few other changes required in manifests. This script
+works "as is" for a regular UPI deployment, but can be edited to reflect your own needs.
+
+For example, if you'd like to use a Virtual Network and subnets already existing in your organization (instead of a having new one
+created specifically for this deployment), you could either edit the `manifests/cluster-infrastructure-02-config.yml` file manually,
+or edit the script to provide their names and the resource group where they exist.
+
+```python
+yamlx['status']['platformStatus']['azure']['networkResourceGroupName'] = "RESOURCE-GROUP-OF-MY-VNET"
+yamlx['status']['platformStatus']['azure']['virtualNetwork'] = "MY-VNET-NAME"
+yamlx['status']['platformStatus']['azure']['controlPlaneSubnet'] = resource_group + "MY-CONTROL-PLANE-SUBNET-NAME"
+yamlx['status']['platformStatus']['azure']['computeSubnet'] = resource_group + "MY-COMPUTE-NODES-SUBNET-NAME"
+```
+
+Finally, run the script to have the changes applies to your manifests:
+
+```sh
+python3 setup-manifests.py $RESOURCE_GROUP
+```
+
+### Remove control plane machines and machinesets
+
+Remove the control plane machines and compute machinesets from the manifests.
+We'll be providing those ourselves and don't want to involve the [machine-API operator][machine-api-operator].
+
+```sh
+rm -f openshift/99_openshift-cluster-api_master-machines-*.yaml
+rm -f openshift/99_openshift-cluster-api_worker-machineset-*.yaml
```

Comment:
TODO: need to verify the file names are correct.

Reply:
Just verified, they're correct.